### PR TITLE
Stabilize libscabbard "receipt-store"

### PR DESCRIPTION
**Pull request by isabeltomb**
Stabilize the "receipt-store" feature in libscabbard by removing it

### CODE REVIEW

File: services/scabbard/libscabbard/Cargo.toml [hunk 1]
```diff
@@ -45,7 +45,7 @@ transact = { version = "0.3.10", features = ["sawtooth-compat"] }
 [dependencies.sawtooth]
 version = "0.6.7"
 default-features = false
-features = ["lmdb-store", "receipt-store", "transaction-receipt-store"]
+features = ["lmdb", "transaction-receipt-store"]
```
Comment (vaporos) [inline services/scabbard/libscabbard/Cargo.toml:48]: This change - "lmdb-store" to "lmdb" looks unintentional.

File: services/scabbard/libscabbard/src/service/factory/factory_database.rs [hunk 2]
```diff
@@ -44,9 +44,6 @@ use crate::store::{
 
 const DEFAULT_LMDB_DIR: &str = "/var/lib/splinter";
 
-#[cfg(feature = "receipt-store")]
-type ScabbardReceiptStore = Arc<RwLock<dyn ReceiptStore>>;
```
Comment (vaporos) [inline services/scabbard/libscabbard/src/service/factory/factory_database.rs:48]: This change looks unrelated to removing the feature.

**Review by vaporos — CHANGES_REQUESTED** [review]

File: services/scabbard/libscabbard/src/service/factory/factory_database.rs (hunk 2 above)
Comment (isabeltomb) [reply services/scabbard/libscabbard/src/service/factory/factory_database.rs:48]: I removed it because the type was originally created so that separate args weren't needed for the two different receipt stores. It was also removed from `factory_static_configuration.rs` but you're right, in this file it isn't directly related so I changed this back

File: services/scabbard/libscabbard/Cargo.toml (hunk 1 above)
Comment (vaporos) [reply services/scabbard/libscabbard/Cargo.toml:48]: Ok, makes sense.

File: services/scabbard/libscabbard/src/service/factory/factory_static_configuration.rs [hunk 3]
```diff
@@ -457,95 +432,65 @@ impl ServiceFactory for ScabbardFactory {
...
-                feature = "receipt-store",
-                not(any(feature = "postgres", feature = "sqlite"))
-            ))]
-            _ => {
-                return Err(FactoryCreateError::Internal(
-                    "receipt-store was enabled, but without any databases".into(),
-                ))
-            }
-        };
+        let (receipt_store, receipt_purge): (Arc<RwLock<dyn ReceiptStore>>, _) =
```
Comment (vaporos) [inline services/scabbard/libscabbard/src/service/factory/factory_static_configuration.rs:435]: This also looks unrelated, removing the use of ScabbardReceiptStore. Should at minimum be a separate commit.

**Review by agunde406 — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]
